app/core/aceUtils.coffee

```diff
@@ -144,7 +144,9 @@ parseUserSnippets = (source, lang, session) ->
 setupCRDT = (key, userName, doc, editor, next) =>
   ydoc = new Y.Doc()
   server = window.location.host
-  url = "ws://#{server}/yjs/websocket/level.session"
+  url = "wss://#{server}/yjs/websocket/level.session"
+  if window.location.protocal != 'https:'
```

Comment:
`protocol`

Also, can you move this logic to a function like `websocketBaseUrl` ?

Reply:
oh good idea